runtime/moonbeam/src/lib.rs

```diff
@@ -1620,7 +1661,7 @@ moonbeam_runtime_common::impl_runtime_apis_plus_common! {
 					// Calculate how much gas this effectively uses according to the existing mapping
 					let effective_gas =
 						<Runtime as pallet_evm::Config>::GasWeightMapping::weight_to_gas(
-							dispatch_info.weight
+							dispatch_info.call_weight
```

Comment:
I think we want to use `total_weight`. Probably it will not make a difference here.
```suggestion
							dispatch_info.total_weight()
```

Reply:
Addressed in https://github.com/moonbeam-foundation/moonbeam/pull/3224/commits/e1a01eeda369a6eb4e4d1b3cb4905916fd3c2ce6